File: 2025/Company/Company.Web/Modules/ServerTypes/Company/CurrencyForm.ts
import { StringEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface CurrencyForm {
    CurrencyCode: StringEditor;
    Name: StringEditor;
}

export class CurrencyForm extends PrefixedContext {
    static readonly formKey = 'Company.Currency';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!CurrencyForm.init)  {
            CurrencyForm.init = true;

            var w0 = StringEditor;

            initFormType(CurrencyForm, [
                'CurrencyCode', w0,
                'Name', w0
            ]);
        }
    }
}

File: Ser2023/Company/Company.Web/Modules/Company/Currency/CurrencyForm.cs
using Serenity.ComponentModel;

namespace Company.Company.Forms;

[FormScript("Company.Currency")]
[BasedOnRow(typeof(CurrencyRow), CheckNames = true)]
public class CurrencyForm
{
    [OneThirdWidth]
    public string CurrencyCode { get; set; }
    [TwoThirdWidth]
    public string Name { get; set; }
}